docs/explanation/high-availability.md

```diff
@@ -0,0 +1,40 @@
+# High availability
+High availability (HA) is a characteristic of a system that aims to ensure an agreed level of operational performance, usually uptime, for a higher than normal period. In this document we will discuss different possible configurations for High availability supported by the `hacluster` subordinate charm.
+
+# Active/active
+An active-active configuration consists of two or more Haproxy units that reply to incoming requests at the same time. Increasing the number of haproxy charm units can be done either during deployment with the `-n` flag or by using the `juju add-unit` command: 
+```
+juju add-unit haproxy -n 2
+```
+
+To configure a single entrypoint for all haproxy units, we can employ a variety of methods, which includes but not limited to:
+* Using a floating IP address
+* DNS round-robin
+* Using a load balancer
+
+Note: Configuration of the entrypoint for the active-active configuration is outside of the scope of responsibility for the haproxy charm.
+
+## DNS round-robin
+DNS round-robin provides a single entrypoint for all haproxy units with no overhead, shifting the responsibility to the DNS server for load balancing purposes. However,  most DNS solutions lack health checks of unhealthy hosts, as well as requiring configuration changes every time there’s a change in the number of haproxy units or their IP addresses.
+
+## Load balancing
```

Comment:
Would adding a note about session persistence (sticky sessions) be helpful?

Reply:
I'm not sure what to put here regarding that, I think we can say that configuring load balancing based on session cookie for example is recommended but it's ultimately up to the operator to configure it for the specific load-balancer that is used. So it seems a bit out of scope for this docs, what do you think?

Reply:
I thought more about a warning that there is nothing in haproxy tied to a specific backend so this should be configured  explicitly or use a different load balancing strategy. wdyt

Reply:
added the note in the load balancing section!